ir/json_generator.h

```diff
@@ -80,11 +80,18 @@ class JSONGenerator {
 
     template<typename T, typename U>
     void generate(const std::pair<T, U> &v) {
-        out << "{" << std::endl << ++indent << "\"first\" : ";
+        ++indent;
+        out << "{" << std::endl;
+        toJSON(v);
+        out << std::endl << --indent << "}";
+    }
+
+    template<typename T, typename U>
+    void toJSON(const std::pair<T, U> &v) {
```

Comment:
I think that the name of this function is misleading; someone may expect it to generate a full JSON object.
You should call it something else, like toJsonObjectBody.

Reply:
I agree, but this name is actually consistent with `Node::toJSON()`: https://github.com/p4lang/p4c/blob/d37330b117dd4bb42e21067a75f67e5fa3a5749c/ir/node.cpp#L27